plugins/conf_maps/lib/api/KalturaConfMaps.php

```diff
@@ -105,17 +110,41 @@ public function validateForInsert($propertiesToSkip = array())
 		parent::validateForInsert($propertiesToSkip);
 	}
 
-	public function validateContent()
+	public function validateAndHandleContent()
 	{
-		$contentArray = json_decode($this->content, true);
-		if(!$contentArray)
+		$content = json_decode($this->content, true);
+		if (is_null($content))
+		{
+			throw new KalturaAPIException(KalturaErrors::CANNOT_PARSE_CONTENT, "Cannot JSON decode content", $this->content);
+		}
+
+		$filter = new KalturaConfMapsFilter();
+		$filter->relatedHostEqual = $this->relatedHost;
+		$filter->nameEqual = $this->name;
+		kApiCache::disableCache();
+		$configurationMap = $filter->getMap(true);
+		$contentToValidate = null;
+		if ($configurationMap)
+		{
+			$existingMapsContent = json_decode($configurationMap->content, true);
+			if (!is_null($existingMapsContent))
+			{
+				$ini = new Zend_Config($existingMapsContent, true);
+				$contentToValidate = iniUtils::arrayToIniString($ini->toArray());
+			}
+		}
+		$contentToValidate .= "\n" . $content;
+		try
 		{
-			throw new KalturaAPIException(KalturaErrors::CANNOT_PARSE_CONTENT , "Cannot JSON decode content"  ,$this->content );
+			$tempIniFile = tempnam(sys_get_temp_dir(), 'TMP_CONF_MAP_');
+			file_put_contents($tempIniFile, $contentToValidate);
+			new Zend_Config_Ini($tempIniFile);
+			unlink($tempIniFile);
 		}
-		$initStr = iniUtils::arrayToIniString($contentArray);
-		if(!parse_ini_string($initStr,true))
+		catch (Exception $e)
 		{
-			throw new KalturaAPIException(KalturaErrors::CANNOT_PARSE_CONTENT, "Cannot parse INI", $initStr);
+			KalturaLog::err($e->getMessage());
```

Comment:
Since the exception will already print the callback , use kalturaLog:warning so we can just read the error message.